src/main/scala/esmeta/analyzer/AbsSemantics.scala

```diff
@@ -53,13 +53,21 @@ class AbsSemantics(
   /** get elapsed time of analyzer */
   def elapsedTime: Long = System.currentTimeMillis - startTime
 
+  /** check reachability of node points */
+  def reachable(np: NodePoint[Node]): Boolean = !apply(np).isBottom
+
+  /** check reachability of return points */
+  def reachable(rp: ReturnPoint): Boolean = !apply(rp).isBottom
+
   /** set of analyzed functions */
-  def analyzedFuncs: Set[Func] =
-    npMap.keySet.map(_.func) ++ rpMap.keySet.map(_.func)
+  def analyzedFuncs: Set[Func] = npMap.keySet.map(_.func) ++ analyzedReturns
 
   /** set of analyzed nodes */
   def analyzedNodes: Set[Node] = npMap.keySet.map(_.node)
 
+  /** set of analyzed nodes */
+  def analyzedReturns: Set[Func] = rpMap.keySet.map(_.func)
```

Comment:
```suggestion
  /** set of returned functions */
  def analyzedReturns: Set[Func] = rpMap.keySet.map(_.func)
```
or other one?

Reply:
Sorry, I merged this PR before applying your suggestion. I applied this suggestion to `adv-ty-refine` branch.